p2p/src/net/default_backend/backend.rs

```diff
@@ -125,10 +130,10 @@ pub struct Backend<T: TransportSocket> {
 
     /// Channel sender for sending events from Peers to Backend; this will be passed to each
     /// Peer upon its creation.
-    peer_event_tx: mpsc::UnboundedSender<(PeerId, PeerEvent)>,
+    peer_event_tx: mpsc::Sender<(PeerId, PeerEvent)>,
```

Comment:
Fixed. Now each `Peer` has a dedicated channel for sending `PeerEvent`s. The channels' receivers are converted to tokio's `ReceiverStream` at the `Backend` and then aggregated into `StreamMap`, so that they can be `select`ed at once.